src/test/unit/auto_tag/test_auto_tag_event.py

```diff
@@ -0,0 +1,276 @@
+from lumigo_tracer.auto_tag import auto_tag_event
+from lumigo_tracer.auto_tag.auto_tag_event import EventAutoTagHandler, AutoTagEvent
+from lumigo_tracer.spans_container import SpansContainer
+from lumigo_tracer.utils import EXECUTION_TAGS_KEY
+
+
+class ExceptionHandler(EventAutoTagHandler):
+    @staticmethod
+    def is_supported(event) -> bool:
+        raise Exception()
+
+    @staticmethod
+    def auto_tag(event):
+        raise Exception()
+
+
+def test_auto_tag_event_is_none():
+    AutoTagEvent.auto_tag_event(event=None)
+
+    assert SpansContainer.get_span().function_span[EXECUTION_TAGS_KEY] == []
+
+
+def test_auto_tag_exception():
+    event = {"a": 1}
+
+    AutoTagEvent.auto_tag_event(event=event, handlers=[ExceptionHandler()])
+
+    assert SpansContainer.get_span().function_span[EXECUTION_TAGS_KEY] == []
+
+
+def test_auto_tag_key_not_in_header(monkeypatch):
+    monkeypatch.setenv("LUMIGO_AUTO_TGA_API_GW_HEADERS", "key_not_exists")
+
+    event = {
```

Comment:
I think it will be nice to extract this variable to a function, just to make the test a bit more readable :)